src/eth/storage/permanent_storage.rs

```diff
@@ -0,0 +1,109 @@
+    async fn maybe_read_account(&self, address: &Address, point_in_time: &StoragePointInTime) -> anyhow::Result<Option<Account>>;
+
+    /// Retrieves an slot from the storage. Returns Option when not found.
+    async fn maybe_read_slot(&self, address: &Address, slot_index: &SlotIndex, point_in_time: &StoragePointInTime) -> anyhow::Result<Option<Slot>>;
+
+    /// Retrieves a block from the storage.
+    async fn read_block(&self, block_selection: &BlockSelection) -> anyhow::Result<Option<Block>>;
+
+    /// Retrieves a transaction from the storage.
+    async fn read_mined_transaction(&self, hash: &Hash) -> anyhow::Result<Option<TransactionMined>>;
+
+    /// Retrieves logs from the storage.
+    async fn read_logs(&self, filter: &LogFilter) -> anyhow::Result<Vec<LogMined>>;
+
+    /// Persists atomically all changes from a block.
+    async fn save_block(&self, block: Block) -> anyhow::Result<(), EthStorageError>;
+
+    /// Persists initial accounts (test accounts or genesis accounts).
+    async fn save_initial_accounts(&self, accounts: Vec<Account>) -> anyhow::Result<()>;
+
+    /// Temporarily stores account changes during block production
+    async fn save_account_changes(&self, block_number: BlockNumber, execution: Execution) -> anyhow::Result<()>;
+
+    /// Resets all state to a specific block number.
+    async fn reset(&self, number: BlockNumber) -> anyhow::Result<()>;
+    /// Enables genesis block.
+    ///
+    /// TODO: maybe can use save_block from a default method.
+    async fn enable_genesis(&self, genesis: Block) -> anyhow::Result<()>;
+
+    // -------------------------------------------------------------------------
+    // Default operations
+    // -------------------------------------------------------------------------
+
+    /// Retrieves an account from the storage. Returns default value when not found.
+    async fn read_account(&self, address: &Address, point_in_time: &StoragePointInTime) -> anyhow::Result<Account> {
+        match self.maybe_read_account(address, point_in_time).await? {
+            Some(account) => Ok(account),
+            None => Ok(Account {
+                address: address.clone(),
+                ..Account::default()
+            }),
+        }
+    }
+
+    /// Retrieves an slot from the storage. Returns default value when not found.
+    async fn read_slot(&self, address: &Address, slot_index: &SlotIndex, point_in_time: &StoragePointInTime) -> anyhow::Result<Slot> {
+        match self.maybe_read_slot(address, slot_index, point_in_time).await? {
+            Some(slot) => Ok(slot),
+            None => Ok(Slot {
+                index: slot_index.clone(),
+                ..Default::default()
+            }),
+        }
+    }
+
+    /// Translates a block selection to a specific storage point-in-time indicator.
+    async fn translate_to_point_in_time(&self, block_selection: &BlockSelection) -> anyhow::Result<StoragePointInTime> {
```

Comment:
This should probably be in `StratusStorage` itself instead of being in this trait.